src/middlewares/index.js

```diff
@@ -4,12 +4,30 @@ function isAdmin(req, res, next) {
     next();
   } else {
     res.status(401);
-    res.json({
-      message: 'Un-Authorized',
-    });
+    const error = new Error('Un-Authorized');
+    next(error);
   }
 }
 
+function notFound(req, res, next) {
+  const error = new Error(`Not Found - ${req.originalUrl}`);
+  res.status(404);
+  next(error);
+}
+
+// eslint-disable-next-line no-unused-vars
+function errorHandler(error, req, res, next) {
```

Comment:
We can destructure the error here to { message, stack } to make the response json much cleaner.